internal/contract/infrastructure_cluster.go

```diff
@@ -61,6 +61,79 @@ func (c *InfrastructureClusterControlPlaneEndpoint) Port() *Int64 {
 	}
 }
 
+// Ready provides access to the status.ready field in an InfrastructureCluster object.
+func (c *InfrastructureClusterContract) Ready() *Bool {
+	return &Bool{
+		path: []string{"status", "ready"},
+	}
+}
+
+// FailureReason provides access to the status.failureReason field in an InfrastructureCluster object. Note that this field is optional.
+func (c *InfrastructureClusterContract) FailureReason() *String {
+	return &String{
+		path: []string{"status", "failureReason"},
+	}
+}
+
+// FailureMessage provides access to the status.failureMessage field in an InfrastructureCluster object. Note that this field is optional.
+func (c *InfrastructureClusterContract) FailureMessage() *String {
+	return &String{
+		path: []string{"status", "failureMessage"},
+	}
+}
+
+// Option 1 for FailureDomains:
+// Pros: Provides granularity for accessing map[string]FailureDomainSpec using intermediate structs, all `path` return
+// types are wrappers for primatives instead of custom structs.
+// Cons: Accessing the map[string]FailureDomainSpec requires knowing the keys ahead of time.
+
+// FailureDomains provides access to the status.failureDomains field in an InfrastructureCluster object. Note that this field is optional.
+func (c *InfrastructureClusterContract) FailureDomains() *InfrastructureClusterFailureDomains {
```

Comment:
Here's one way we could implement `FailureDomains()` which is a map[string]FailureDomainSpec as outlined here. I implemented two options in the PR, in this one we'd need to know the key name ahead of time to use `FailureDomainSpec`.